buildSrc/src/main/kotlin/org/gradle/plugins/ideconfiguration/IdeConfigurationPlugin.kt

```diff
@@ -0,0 +1,448 @@
+            "keyword" to "Copyright",
+            "allowReplaceKeyword" to "",
+            "myName" to "ASL2",
+            "myLocal" to "true")
+        val copyrightManager = root.select("component[name=CopyrightManager]").first()
+        copyrightManager.attr("default", "ASL2")
+        copyrightManager.createOrEmptyOutChildElement("copyright").let {
+            options.forEach { name, value ->
+                it.attr(name, value)
+            }
+        }
+    }
+
+    private fun configureCompilerConfiguration(root: Element) {
+        val compilerConfiguration = root.select("component[name=CompilerConfiguration]").first()
+        compilerConfiguration.createOrEmptyOutChildElement("excludeFromCompile")
+        compilerConfiguration.removeExistingChildElement("option[name=BUILD_PROCESS_HEAP_SIZE]").
+            appendElement("option").
+            attr("name", "BUILD_PROCESS_HEAP_SIZE").
+            attr("value", "2048")
+    }
+
+    private fun getGradleRunnerConfiguration(runnerName: String, commandLine: String): String {
+        return """
+            <configuration default="false" name="$runnerName" type="Application" factoryName="Application">
+                <extension name="coverage" enabled="false" merge="false" />
+                <option name="MAIN_CLASS_NAME" value="org.gradle.testing.internal.util.GradlewRunner" />
+                <option name="VM_PARAMETERS" value="" />
+                <option name="PROGRAM_PARAMETERS" value="$commandLine" />
+                <option name="WORKING_DIRECTORY" value="file://${'$'}PROJECT_DIR$" />
+                <option name="ALTERNATIVE_JRE_PATH_ENABLED" value="false" />
+                <option name="ALTERNATIVE_JRE_PATH" value="" />
+                <option name="ENABLE_SWING_INSPECTOR" value="false" />
+                <option name="ENV_VARIABLES" />
+                <option name="PASS_PARENT_ENVS" value="true" />
+                <module name="internalTesting" />
+                <envs>
+                    <env name="TERM" value="xterm" />
+                </envs>
+                <method>
+                    <option name="Make" enabled="false" />
+                </method>
+            </configuration>"""
+    }
+
+    @Suppress("UNCHECKED_CAST")
+    private fun getDefaultJunitVmParameter(docsProject: Project): String {
+        val rootProject = docsProject.rootProject
+        val releaseNotesMarkdownTask = docsProject.tasks["releaseNotesMarkdown"] as PegDown
+        val releaseNotesTask = docsProject.tasks["releaseNotes"] as Copy
```

Comment:
:x: no need to cast here
```kotlin
val releaseNotesMarkdown: PegDown by docsProject.tasks
val releaseNotes: Copy by docsProject.tasks
```
And remove `@Suppress("UNCHECKED_CAST")`

Reply:
I get a warning without it: w: /usr/local/hans/gradle-source/buildSrc/src/main/kotlin/org/gradle/plugins/ideconfiguration/IdeConfigurationPlugin.kt: (372, 92): Unchecked cast: Any? to List<String>`

Reply:
Right! I missed the two project properties casts.
We'll be able to express this better once https://github.com/gradle/kotlin-dsl/issues/626 is done.